8.observability/README.md

```diff
@@ -0,0 +1,126 @@
+3. [Configure Redis](https://github.com/dapr/docs/tree/master/howto/configure-redis) as a state store for Dapr.
+
+## Configure Dapr tracing in the cluster
+
+Review *./deploy/tracing.yaml*:
+
+```yaml
+apiVersion: dapr.io/v1alpha1
+kind: Configuration
+metadata:
+  name: tracing
+spec:
+  tracing:
+    enabled: true
+    expandParams: true
+    includeBody: true
+```
+
+This configuration definition will enable Dapr to do tracing. Deploy the configuration by running:
+
+`kubectl apply -f ./deploy/tracing.yaml`
+
+## Deploy Zipkin to the cluster and set it as the tracing provider
+
+In this sample Zipkin is used as the underlying tracing system. Examine [*./deploy/zipkin.yaml*](./deploy/zipkin.yaml) and see how it includes three sections:
+
+1. A **Deployment** for Zipkin using the *openzipkin/zipkin* docker image.
+2. A **Service** which will expose Zipkin internally as a ClusterIP in Kubernetes.
+3. A **Component** that defines Zipkin as the tracing provider for Dapr.
+
+Deploy to your cluster by running:
+
+`kubectl apply -f ./deploy/zipkin.yaml`
+
+Now that Zipkin is deployed, you can access the Zipkin UI by creating a tunnel to the internal Zipkin service you just created by running:
+
+`kubectl port-forward svc/zipkin 9411:9411`
+
+On your browser go to [http://localhost:9411](http://localhost:9411). You should be able to see the Zipkin dashboard.
+
+## Instrument the application for tracing and deploy it
+
+To instrument a service for tracing with Dapr, no code changes are required - Dapr handles all of the tracing using the Dapr side-car. All that is needed is to add the Dapr annotation for tracing in the deployment yaml along with the other Dapr annotations. The annotation looks like this:
+
+```yaml
+...
+annotations:
+...
+    dapr.io/config: "tracing"
+...
+ ```
+
+For this sample, no action is needed since tracing is already enabled for every service in the distributed calculator app. You can find the annotation in each one of the calculator yaml files - For example review the yaml file for the calculator front end service [here](https://github.com/dapr/samples/blob/master/3.distributed-calculator/deploy/react-calculator.yaml#L36).
+
+Note you did not introduce any dependency on Zipkin into the calculator app code or deployment yaml files.
+
+Now deploy the distributed calculator application to your cluster following the instructions found in sample [3.distributed-calculator](https://github.com/dapr/samples/blob/master/3.distributed-calculator/README.md) and have the application running on a Kubernetes cluster. See that you can browse to the calculator UI via a browser.
+
+> **Note:** If the distributed calculator is already running on your cluster you will need to restart it for the tracing enablement to take effect. You can do so by running:
+
+> `kubectl rollout restart deployment/addapp deployment/calculator-front-end deployment/divideapp  deployment/multiplyapp  deployment/subtractapp`
```

Comment:
It might be simpler to just 

kubectl delete -f .
kubectl create -f .

from /deploy.  Either way.

Reply:
That's what I initially did but for me I found it to be less preferable since delete + create requires a new IP to be allocated and takes more time, also my Redis config would get messed up when I was doing these one after the other and didn't notice there's a redis.yaml in the sample that needs to be updated.

There is however a slightly shorter way to do the restart: 
`kubectl rollout restart deployment addapp calculator-front-end divideapp multiplyapp subtractapp`

I also like this because it's only one line to copy/paste :)

Reply:
Yep was just a suggestion.